sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -3334,9 +3337,6 @@ class Analyzer(override val catalogManager: CatalogManager) extends RuleExecutor
           v2Write
         }
 
-      case u: UpdateTable if !u.skipSchemaResolution && u.resolved =>
```

Comment:
I have removed this resolution but the logic is same: runtime null checks, varchar/char length checks, etc.